app/models/credential_factory.rb

```diff
@@ -0,0 +1,49 @@
+module CredentialFactory
+  class << self
+    def values factory_resource
+      provider_annotation = factory_resource.credential_factory_provider
+      raise ArgumentError, %Q(Annotation "credential-factory/provider" not found on #{factory_resource.id.inspect}) unless provider_annotation
+      require "credential_factory/#{provider_annotation.value.underscore}"
```

Comment:
This will fail with 500 if the plugin is not found. I'm not sure what the correct code would be -- it's not exactly a client error, so suggests 5xx. Maybe 404, with an informative message? An informative message would be nice regardless of the code.